internal/sequencers/espresso/espresso_listener.go

```diff
@@ -92,9 +93,33 @@ func (e EspressoListener) watchNewTransactions(ctx context.Context) error {
 				if err != nil {
 					return err
 				}
+
+				// assume the following encoding
+				// transaction = JSON.stringify({
+				//		 	signature,
+				//		 	typedData: btoa(JSON.stringify(typedData)),
+				//		 })
+				msgSender, typedData, err := ExtractSigAndData(string(transaction))
+				if err != nil {
+					return err
+				}
+				fmt.Println(msgSender)
+				// type EspressoMessage struct {
+				// 	nonce   uint32 `json:"nonce"`
+				// 	payload string `json:"payload"`
+				// }
+				nonce, ok1 := typedData.Message["nonce"].(uint32)
+				payload, ok2 := typedData.Message["payload"].(string)
+				if !ok1 || !ok2 {
+					slog.Debug("type assertion error ")
```

Comment:
Should we return an error when the cast fails?

Reply:
Yeah. Makes sense